server/blob/blob-objectstorage/src/main/java/org/apache/james/blob/objectstorage/ObjectStorageBlobsDAOBuilder.java

```diff
@@ -46,10 +48,16 @@ public ObjectStorageBlobsDAOBuilder blobIdFactory(BlobId.Factory blobIdFactory)
         return this;
     }
 
+    public ObjectStorageBlobsDAOBuilder payloadCodec(PayloadCodec payloadCodec) {
+        this.payloadCodec = payloadCodec;
+        return this;
+    }
+
     public ObjectStorageBlobsDAO build() {
         Preconditions.checkState(containerName != null);
         Preconditions.checkState(blobIdFactory != null);
-        return new ObjectStorageBlobsDAO(containerName, blobIdFactory, supplier.get());
+
+        return new ObjectStorageBlobsDAO(containerName, blobIdFactory, supplier.get(), payloadCodec);
```

Comment:
I think that `payloadCode` is not optional, can you introduce a precondition for that?

Reply:
payloadCodec is optional in the builder because there is a valid default value which is applied in the constructor see line 37 :)

Reply:
Ok, maybe it's more simple to have an `Optional<PayloadCodec>` in the builder; then in the `build()` method get the default value if none:
`payloadCodec.orElse(PaylocaCode.DEFAULT_CODE)`.
That way dealing with users/default value is done in only one place.